File: tauri-app/src/lib/tauri/query.ts
import { invoke } from "@tauri-apps/api/core";

import type {
  AnalyticsQueryInput,
  AnalyticsResponse,
  LogsQueryInput,
  LogsQueryResponse,
  OverviewResponse,
} from "../../types/query";

export function queryOverview(accountId: string) {
  return invoke<OverviewResponse>("query_overview", { accountId });
}

export function queryLogs(input: LogsQueryInput) {
  return invoke<LogsQueryResponse>("query_logs", { input });
}

export function queryAnalytics(input: AnalyticsQueryInput) {
  return invoke<AnalyticsResponse>("query_analytics", { input });
}

File: tauri-app/src-tauri/src/commands/query.rs
use tauri::State;

use crate::{
    app_state::AppState,
    services::query_service::{
        AnalyticsQueryInput, AnalyticsResponse, LogsQueryInput, LogsQueryResponse, OverviewResponse,
        QueryService,
    },
};

#[tauri::command]
pub fn query_overview(
    state: State<'_, AppState>,
    account_id: String,
) -> Result<OverviewResponse, String> {
    QueryService::query_overview(state.inner(), &account_id).map_err(|error| error.to_string())
}

#[tauri::command]
pub fn query_logs(
    state: State<'_, AppState>,
    input: LogsQueryInput,
) -> Result<LogsQueryResponse, String> {
    QueryService::query_logs(state.inner(), input).map_err(|error| error.to_string())
}

#[tauri::command]
pub fn query_analytics(
    state: State<'_, AppState>,
    input: AnalyticsQueryInput,
) -> Result<AnalyticsResponse, String> {
    QueryService::query_analytics(state.inner(), input).map_err(|error| error.to_string())
}
